### PR TITLE
UCP/UCT/UCS/UCM/GTEST: Fix coverity issues.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9773/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:bgate:retest

File: src/ucp/core/ucp_request.inl [hunk 1]
```diff
@@ -166,6 +166,7 @@ ucp_request_try_send(ucp_request_t *req, ucs_status_t *req_status,
 {
     ucs_status_t status;
 
+    /* coverity[address_free] */
```
Comment (yosefe) [inline src/ucp/core/ucp_request.inl]: can you pls comment what was the issue here?

Comment (alinask) [reply src/ucp/core/ucp_request.inl]: *(*req).send.uct.func" frees address of "(*req).send.uct when func is resolved to test_uct_pending::pending_send_op_ok.
but then, if the req is released, then UCS_OK should return and then req isn't accessed anymore.

**Comment by alinask** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9768/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9778/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9787/ for details (Mellanox internal link).

File: src/ucp/core/ucp_request.inl (hunk 1 above)
Comment (yosefe) [reply src/ucp/core/ucp_request.inl]: let's add comment here that coverity wrongly resolves (*req).send.uct.func to test_uct_pending::pending_send_op_ok

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9808/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9814/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9823/ for details (Mellanox internal link).

**Pull request merged**